File: java/src/test/java/problems1_100/PermutationsTest.java
package problems1_100;

import java.util.Arrays;
import java.util.stream.Collectors;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class PermutationsTest extends SampledTest {
    @Test
    public void test() {
        for (Permutations solution : Permutations.values()) {
            for (PermutationsSample sample : getSamples(PermutationsSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.permute(sample.input.nums))
                    .containsExactlyElementsIn(
                        Arrays
                            .stream(sample.output)
                            .map(ints ->
                                Arrays
                                    .stream(ints)
                                    .boxed()
                                    .collect(Collectors.toList())
                            ).collect(Collectors.toList())
                    ).inOrder();
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/PermutationsTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class PermutationsTest : SampledTest() {
    @Test
    fun test() =
        Permutations.entries.forEach { solution ->
            getSamples(Array<PermutationsSample>::class.java).forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.permute(input.nums))
                    .containsExactlyElementsIn(output.map { it.toList() })
            }
        }
}
